### ISSUE TITLE
Globals vairables URL

### ISSUE THREAD

**Issue by eloyucu** [postmanlabs/newman#1730]
Hi everyone.

I am sorry to make this low level question, but I didn´t find the solution by myself. Where to find the url for global variables in a workspace?? 

Thanks.

**Comment by sivcan** [comment]
@eloyucu Sorry if I misunderstood your question, did you mean the Docs link for globals?
If yes then here: https://www.getpostman.com/docs/v6/postman/environments_and_globals/manage_globals

Or if you've some other issue in using the globals can you elaborate more?
Possible record a video / GIF and share your OS details & your Postman App version.

**Comment by eloyucu** [comment]
Hi @sivcan 

No I mean... I have a workspace with its own set of global variables, to use them on newman I have to save them as a json file, and everytime I want to change something on those variables I need to download again, overwrite the old json file, commit and push. 

On this repo I have seen this:

>-g <source>, --globals <source>
>Specify the file path or URL for global variables. Global variables are similar to environment variables but have a lower precedence and can be overridden by environment variables having the same name.

So I want to use the url instead the path for the json file. Or... does the URL means that you have to host your own globals file and use you own URL??

Thanks.

**Comment by codenirvana** [comment]
@eloyucu Using URL you can fetch collection/environment from [Postman API](https://docs.api.getpostman.com) or from an external source where your data is hosted.

Check [Using Newman with the Postman API](https://github.com/postmanlabs/newman#using-newman-with-the-postman-api) section in README.

However, Global environment can't be fetched via API yet.
For now, you should either fetch the variables using the environment variables and use the API endpoint to avoid modifying the exported file everytime or,
Use the `--global-var` option to pass a global variable.

**Comment by eloyucu** [comment]
Hi @codenirvana. 
I prefer continue using the `--globals` flag to pass it the path to the json file with all the globals vars. 

But it could be great can use a postman´s url for globals of a workspace that allows the auto synchronization, instead use the static json file.

Thanks.

**Comment by codenirvana** [comment]
@eloyucu I completely agree with your point but, global variables can't be fetched via API yet.

Marking this as a feature request. Will update you once its ready.

**Comment by tyrantrook** [comment]
If this is still not ready, it would be immensely useful to me to have it implemented.

**Comment by Christian-Marc** [comment]
> @eloyucu Using URL you can fetch collection/environment from [Postman API](https://docs.api.getpostman.com) or from an external source where your data is hosted.
> 
> Check [Using Newman with the Postman API](https://github.com/postmanlabs/newman#using-newman-with-the-postman-api) section in README.
> 
> However, Global environment can't be fetched via API yet.
> For now, you should either fetch the variables using the environment variables and use the API endpoint to avoid modifying the exported file everytime or,
> Use the `--global-var` option to pass a global variable.

Is it possible to pass in global variables in the newman.run(option) object

**Comment by eloyucu** [comment]
Thanks @Christian-Marc

**Comment by Christian-Marc** [comment]
Sorry, but I'm trying to at global variables via the newman command-line:
> newman.run({}) 

but they are not picked-up during running. Where can I find more instructions to use this construction? Do I need to add something inside the collection?

**Comment by abhilash-id123** [comment]
I have defined request URL as a global variable but while execution it seems that Newman did not detect the same.
Any solution guys?

**Comment by molteninjabob** [comment]
Any progress on this? I've run into the same issue where I'd really like to reference the global variables via url.

**Comment by XeniyaShirinova** [comment]
Any progress on this? I've run into the same issue where I'd really like to reference the global variables via url.

**Comment by eloyucu** [comment]
Finally I added a json file with my global config to the server, and then I run newman using the flag `--globals` and passing the path to the json file.

Sorry, I didn't find any other way to do this.

**Comment by ashish-infogain** [comment]
Any progress on this? I've run into the same issue where I'd really like to reference the global variables via URL. We want to implement the global variable via URL thing because it keeps up synchronizing with the collection.

**Comment by gunesmes** [comment]
Same here, this would really be beneficial for me.

**Comment by deepakaggarwal7** [comment]
While collections and environments are live, you manually have to download and pass global variables json file which is a pain. This would be a super useful feature to have.

**Comment by anhhanuman** [comment]
+1 please
